groovy/gradle/src/org/netbeans/modules/gradle/NbGradleProjectFactory.java

```diff
@@ -47,24 +47,31 @@
 
     @Override
     public boolean isProject(FileObject dir) {
-        boolean ret = FileUtil.toFile(dir) != null;
-        if (ret) {
-            FileObject pom = dir.getFileObject("pom.xml"); //NOI18N
-            if ((pom != null) && pom.isData() && GradleSettings.getDefault().isPreferMaven()) {
-                ret = false;
-            } else {
-                File suspect = FileUtil.toFile(dir);
-                GradleFiles files = new GradleFiles(suspect);
-                if (!files.isRootProject()) {
-                    Boolean inSubDirCache = GradleProjectCache.isKnownSubProject(files.getRootDir(), suspect);
-                    ret = inSubDirCache != null ? inSubDirCache : files.isProject();
-                } else {
-                    ret = true;
-                }
+        return isProjectCheck(dir, GradleSettings.getDefault().isPreferMaven());
```

Comment:
Well in this case, I'd rather like an 
if (preferMaven && mavenProjectCheck(...)) return false;
<Gradle Project check>
kind of implementation.

Reply:
What exactly do you want? I don't understand.